src/lib.rs

```diff
@@ -138,7 +138,20 @@ where
     use cipher::StreamCipherSeek;
 
     let mut nonce_block = GenericArray::default();
-    nonce_block[..nonce.len()].copy_from_slice(nonce);
+    if nonce.len() == 12 {
+      nonce_block[..nonce.len()].copy_from_slice(nonce);
+    } else {
+      let mut ghash = GHash::new(&ghash_key);
+      ghash.update_padded(nonce);
+      ghash.update_padded(&(8 * nonce.len() as u128).to_be_bytes());
+      nonce_block.copy_from_slice(&ghash.finalize());
+      for i in nonce_block.iter_mut().rev() {
+        *i = i.wrapping_sub(1);
+        if *i != 0xff {
+          break;
+        }
```

Comment:
Can you add a comment linking to the spec page for this algorithm?

Reply:
Added comment with a link to the spec pdf.

BTW this for-loop is not something specified in the spec, but a workaround to align it with CTR implementation used below. The CTR impl seems incrementing the counter block once before the initial calculation. So we need to subtract 1 from it. (The nonce for the case length is 96 is also decremented 1 from what spec says. (We pass `IV || 0^32` instead of `IV || 0^31 || 1`)